### PR TITLE
[W5.11][W14-A1] K Muruges

### PR DESCRIPTION
**Pull request by muruges95**
Second feature for exercise 5.11; made the add and delete operations also show the new list of people in the addressbook. The code, user guide and all tests have also been updated for this feature.

There is also the creation of a Sort command that is also reflected in this pull request but ignore them as I made that feature mistakenly on the master branch before I branched off and worked on this feature. That one is part of my first pull request for the first feature created for exercise 5.11

### CODE REVIEW

File: test/java/seedu/addressbook/commands/AddCommandTest.java [hunk 1]
```diff
@@ -5,11 +5,7 @@
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
```
Comment (shanwpf) [inline test/java/seedu/addressbook/commands/AddCommandTest.java:8]: I think you should change this to individual imports

File: src/seedu/addressbook/commands/DeleteCommand.java [hunk 2]
```diff
@@ -30,7 +32,8 @@ public CommandResult execute() {
         try {
             final ReadOnlyPerson target = getTargetPerson();
             addressBook.removePerson(target);
-            return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, target));
+            List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
+            return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, target), allPersons);
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/DeleteCommand.java:36]: Did you update the header comment to reflect this behavior?

File: src/seedu/addressbook/common/Messages.java [hunk 3]
```diff
@@ -15,4 +15,5 @@
             "java seedu.addressbook.Main [STORAGE_FILE_PATH]";
     public static final String MESSAGE_WELCOME = "Welcome to your Address Book!";
     public static final String MESSAGE_USING_STORAGE_FILE = "Using storage file : %1$s";
+    public static final String MESSAGE_STORAGE_FILE_CANNOT_OPEN = "An error has occurred: The storage file is currently read-only.";
```
Comment (okkhoy) [inline src/seedu/addressbook/common/Messages.java:18]: Can wrap this line for better readability.

File: src/seedu/addressbook/data/AddressBook.java [hunk 4]
```diff
@@ -114,6 +114,11 @@ public UniquePersonList getAllPersons() {
         return new UniquePersonList(allPersons);
     }
 
+    /**
+     * Sorts all persons by alphabetical order of their name
+     */
+    public void sortAllPersons(){ allPersons.sort(); }
```
Comment (okkhoy) [inline src/seedu/addressbook/data/AddressBook.java:120]: 1. Please take a look at the coding standards and see how to write methods with a single line of code.
2. Need space between `) {`

File: src/seedu/addressbook/parser/Parser.java [hunk 5]
```diff
@@ -11,17 +11,8 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (okkhoy) [inline src/seedu/addressbook/parser/Parser.java:14]: Don't import everything using *; import the necessary classes explicitly.

File: test/java/seedu/addressbook/commands/DeleteCommandTest.java [hunk 6]
```diff
@@ -111,6 +111,9 @@ private void assertCommandBehaviour(DeleteCommand deleteCommand, String expected
 
         assertEquals(expectedMessage, result.feedbackToUser);
         assertEquals(expectedAddressBook.getAllPersons(), actualAddressBook.getAllPersons());
+        if (expectedMessage.equals(DeleteCommand.MESSAGE_DELETE_PERSON_SUCCESS)){
```
Comment (okkhoy) [inline test/java/seedu/addressbook/commands/DeleteCommandTest.java:114]: need space between `) {`

**Comment by okkhoy** [comment]
@muruges95 
Some comments added. Please ack & close the PR after reading comments.

File: src/seedu/addressbook/parser/Parser.java (hunk 5 above)
Comment (muruges95) [reply src/seedu/addressbook/parser/Parser.java:14]: Alright noted.

**Comment by muruges95** [comment]
Alright noted.

**Pull request closed without merging**